frontend/packages/dev-console/src/components/edit-application/EditApplication.tsx

```diff
@@ -53,7 +55,7 @@ const EditApplication: React.FC<EditApplicationProps> = ({
     if (values.build.strategy) {
       const imageStream =
         values.image.selected && builderImages ? builderImages[values.image.selected].obj : null;
-      if (pageHeading === CreateApplicationFlow.JarUpload) {
+      if (pageHeading === i18n.t(EditFlowPageTitle[CreateApplicationFlow.JarUpload])) {
```

Comment:
This is within the scope of a Component, please use the `t` from `useTranslation`. The main goal is not to pass `t` to functions. Hooks are defined in the component, anything within "it's window" should (for code simplicity) just use the hook `t`.

Reply:
We should avoid comparing titles to enum values and translate the enum values here already. We should translate only values which we render to the user.

If your getPageHeading (think of it as "getApplicationFlow" util function) returns a typed enum it's more clear that this checks should stay at they are before.

(IMHO: To make this more clear you can check `pageHeading` here to `applicationFlow` and remove the i18n.t calls here completly.)

```tsx
const applicationFlow = getApplicationFlow(buildStrategy, buildSourceType);
```